client/src/main/java/io/pravega/client/EventStreamClientFactory.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.client;
+
+import io.pravega.client.netty.impl.ConnectionFactoryImpl;
+import io.pravega.client.stream.EventRead;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.Position;
+import io.pravega.client.stream.ReaderConfig;
+import io.pravega.client.stream.ReaderGroup;
+import io.pravega.client.stream.Serializer;
+import io.pravega.client.stream.TransactionalEventStreamWriter;
+import io.pravega.client.stream.impl.ClientFactoryImpl;
+import io.pravega.client.stream.impl.ControllerImpl;
+import io.pravega.client.stream.impl.ControllerImplConfig;
+import lombok.val;
+
+/**
+ * Used to create Writers, Readers, and Synchronizers operating on a stream.
+ * <p>
+ * Events that are written to a stream can be read by a reader. All events can be processed with
+ * exactly once semantics provided the reader has the ability to restore to the correct position
+ * upon failure. See {@link EventRead#getPosition()}
+ * <p>
+ * A note on ordering: Events inside of a stream have a strict order, but may need to be divided
```

Comment:
`strict order` is not a term I know. It might be better to elaborate on how it is strict.